### PR TITLE
Fixes #2977

### LINKED ISSUE #2977: Auto-Enchanter has strict slot order
## :round_pushpin: Description (REQUIRED)

The author enchanter does not work if the item is on the left (first) slot

## :bookmark_tabs: Steps to reproduce the Issue (REQUIRED)

1. Place auto enchanter
2. Insert an item (such as a pickaxe) into the left (first) slot
3. Insert an enchanted book (eg unbreaking) into the right (second) slot
4. Nothing happens
If the enchanted book is placed in the left (first) slot and the item in the right (second), the auto enchanter works.
Video: https://youtu.be/NPX6k2nsg-Q

## :bulb: Expected behavior (REQUIRED)

The auto enchanter will work if there is an item in the right (first) slot and an enchanted book in the left (second) slot

## :scroll: Server Log

N/A

## :open_file_folder: /error-reports/ Folder

N/A

## :compass: Environment (REQUIRED)

image

### PR DESCRIPTION
**Pull request by martinbrom**
## Description

Auto Enchanter now considers both combinations of enchanted book and item (item in first or second slot).

## Proposed changes

The machine now tries to check if the second combination could possibly be valid instead of exiting immediately as before.
Uses `addUnsafeEnchantments` instead of `addUnsafeEnchantment` to skip the loop over map entries.
Uses `isEmpty` and `map.size()` instead of a separate amount variable.

## Related Issues (if applicable)

Resolves #2977 

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/enchanting/AutoEnchanter.java [hunk 1]
```diff
@@ -53,7 +53,7 @@ protected MachineRecipe findNextRecipe(BlockMenu menu) {
 
             // Check if the item is enchantable
             if (!isEnchantable(item)) {
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/enchanting/AutoEnchanter.java:55]: I just noticed that `isEnchantable` is missing an air check but that's just a side note.

**Review by TheBusyBiscuit — APPROVED** [review]
LGTM

Don't even know why we had a seperate `amount` variable there. Parts of this class are over 5 years old...

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/enchanting/AutoEnchanter.java (hunk 1 above)
Comment (martinbrom) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/enchanting/AutoEnchanter.java:55]: Do you want me to add that or can it be ignored?

Comment (TheBusyBiscuit) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/enchanting/AutoEnchanter.java:55]: You can add it if you want but can also ignore it and we will add it later.
I approved it, so no explicit actions are expected nor required :)

**Review by WalshyDev — APPROVED** [review]

**Pull request merged**